assignment3/WordFinderTest.java

```diff
@@ -0,0 +1,42 @@
+package com.katermar.assignment3;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+/**
+ * Created by katermar on 6/9/2017.
+ */
+public class WordFinderTest {
+
+    public static void main(String args[]) {
+        char grid[][] = {
+                {'A', 'A', 'R'},
+                {'T', 'C', 'D'}
+        };
+        String dictArray[] = {"CAR", "CARD", "CART", "CAT"};
+//        String prefixArray[] = {"C", "CA", "CAR", "CARD", "CART", "CAT"};
+        Dictionary dictionary = new Dictionary(dictArray);
+        String expectedResultArray[] = {"CAR", "CARD", "CAT"};
+
+        testWordFinder(new HashSet<String>(Arrays.asList(expectedResultArray)), dictionary, grid);
+        testWordFinder(new HashSet<String>(), dictionary, grid);
+        char emptyGrid[][] = {{}, {}};
+        testWordFinder(new HashSet<String>(Arrays.asList(expectedResultArray)), dictionary, emptyGrid);
+    }
+
+    public static void testWordFinder(HashSet<String> expectedResult, Dictionary dictionary, char grid[][]) {

```

Comment:
If you define your own testing mechanism it is useful to have tests tagged somehow and print this tag as part of the success/failure message, so that it is trivial to find out which test case failed.
It might not be a problem for three test cases, but if you have hundreds of them and you face a stream of multiple pass/fail messages that you then have to manually count to map them to test cases that need to be fixed.

Could you please add something like test name to the method and print it together with results?